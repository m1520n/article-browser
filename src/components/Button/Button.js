import React from 'react';
import PropTypes from 'prop-types';

import { ButtonWrapper } from './Button.style';

const propTypes = {
  type: PropTypes.oneOf(['primary', 'secondary']),
  size: PropTypes.oneOf(['small', 'regular', 'big']),
  disabled: PropTypes.bool,
  block: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
  handleClick: PropTypes.func,
};

const defaultProps = {
  type: 'primary',
  size: 'regular',
  disabled: false,
  block: false,
  handleClick: () => {},
};

const Button = ({
  children, handleClick, type, size, disabled, block,
}) => (
  <ButtonWrapper type={type} size={size} disabled={disabled} block={block} onClick={handleClick}>
    {children}
  </ButtonWrapper>
);

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
