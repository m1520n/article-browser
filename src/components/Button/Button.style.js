import styled from 'styled-components';

import { fonts, colors } from '../../styles/variables';

const sizes = {
  small: {
    width: '22.5rem',
    height: '3.2rem',
  },
  regular: {
    width: '33rem',
    height: '4.8rem',
  },
  big: {
    width: '22.5rem',
    height: '6.4rem',
  },
};

export const ButtonWrapper = styled.button`
  border: none;
  cursor: pointer;
  font-family: ${fonts.arialMT};
  font-size: 1.4rem;
  line-height: ${({ size }) => sizes[size].height};
  width: ${({ size }) => sizes[size].width};
  ${({ block }) => block && 'width: 100%;'}
  height: ${({ size }) => sizes[size].height};
  background-color: ${({ type }) => (type === 'primary' ? colors.royalBlue : colors.white)};
  color: ${({ type }) => (type === 'primary' ? colors.white : colors.royalBlue)};
  border: 1px solid ${colors.royalBlue};
  ${({ disabled }) => disabled && 'pointer-events: none;'}

  :hover {
    background-color: ${({ type }) => (type === 'primary' ? colors.sapphire : colors.royalBlue)};
    border: 1px solid ${({ type }) => (type === 'primary' ? colors.sapphire : colors.royalBlue)};
    color: ${colors.white};
  }

  /* TODO: Should it use the outline for accesibility?
  /* :focus {
    outline:0;
  } */
`;

export default {
  ButtonWrapper,
};
