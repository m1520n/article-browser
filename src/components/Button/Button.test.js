import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';
import { ButtonWrapper } from './Button.style';

test('Button is rendered correctly', () => {
  const component = renderer.create(<Button>TEST BUTTON</Button>);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should invoke handleClick callback when button is clicked', () => {
  const handleClickMock = jest.fn();
  const component = renderer.create(<Button handleClick={handleClickMock}>TEST BUTTON</Button>);

  const button = component.root.findByType(ButtonWrapper);
  const mockEvent = { preventDefault: () => {} };
  button.props.onClick(mockEvent);
  expect(handleClickMock).toHaveBeenCalledWith(mockEvent);
});
