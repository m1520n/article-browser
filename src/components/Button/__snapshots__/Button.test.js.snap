// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is rendered correctly 1`] = `
<button
  className="sc-bdVaJa dfapiH"
  disabled={false}
  onClick={[Function]}
  size="regular"
  type="primary"
>
  TEST BUTTON
</button>
`;
